service/local/segmentFetcher.go

```diff
@@ -0,0 +1,83 @@
+package local
+
+import (
+	"bytes"
+	"crypto/sha1"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/splitio/go-split-commons/v4/dtos"
+	"github.com/splitio/go-split-commons/v4/service"
+	"github.com/splitio/go-toolkit/v5/logging"
+)
+
+// FileSegmentFetcher struct fetches segments from a file
+type FileSegmentFetcher struct {
+	segmentDirectory string
+	addedLastHash    []byte
```

Comment:
you should track a map of segments hash and you don't need to have two. 
map[string, []byte]  where the key is the segmentName

Reply:
Added a map for sha.